Rubberduck.VBEEditor/QualifiedModuleName.cs

```diff
@@ -33,40 +35,27 @@ public static string GetProjectId(IReference reference)
 
         public QualifiedModuleName(IVBProject project)
         {
-            Component = null;
             _componentName = null;
             ComponentType = ComponentType.Undefined;
             _projectName = project.Name;
             ProjectPath = string.Empty;
             ProjectId = GetProjectId(project);           
-            ContentHashCode = 0;
         }
 
         public QualifiedModuleName(IVBComponent component)
         {
             ComponentType = component.Type;
-            Component = component;
             _componentName = component.IsWrappingNullReference ? string.Empty : component.Name;
 
-            ContentHashCode = 0;
-            if (!Component.IsWrappingNullReference)
+            using (var components = component.Collection)
             {
-                using (var module = Component.CodeModule)
+                using (var project = components.Parent)
                 {
-                    ContentHashCode = module.CountOfLines > 0
-                        ? module.GetLines(1, module.CountOfLines).GetHashCode()
-                        : 0;
+                    _projectName = project == null ? string.Empty : project.Name;
+                    ProjectPath = string.Empty;
+                    ProjectId = GetProjectId(project);
                 }
             }
-
-            IVBProject project;
-            using (var components = component.Collection)
-            {
-                project = components.Parent;
```

Comment:
wouldn't that have been a bad idea???

Reply:
What exactly do you mean?

Reply:
The old code leaks a reference obtained from an `IDisposable` to outside of the using scope. At least in Java code this generally means that the reference becomes unusable, since the resource it referred to has been disposed.

Usually that'd either result in an `IllegalStateException` or some other form of unintended behaviour. Only pertains to the old code, though. The new code doesn't leak a reference.

Reply:
The thing is that the COM wrappers do not manage what they return. They just new up things the receiver has to dispose afterwards.